### PR TITLE
refactor: address relaying review comments

### LINKED ISSUE #1838: [Relayer] Fix comments in the epic PR
Please check the comments in the PR: #1776. They should be addressed retroactively.

### PR DESCRIPTION
**Pull request by iamacook**
## What it solves

Resolves #1838 (comments from #1776)

## How this PR fixes it

- `useRelaysBySafe` and `useLeastRemainingRelays` now return the entire to reduce requests
- Chain information in `SponsoredBy` is now dynamic
- `useWalletCanRelay` now logs the error correctly
- Tracking has been moved to `useTxTracking`

## How to test it

- The number of remaining relays are correctly shown in the dashboard, execution modal and Safe creation (taking multiple owner rate limits into account).
- The chain name and logo are correctly shown in the dashboard widget.
- The "Relayed execution" event is dispatched when a transaction is relayed.

## TODO

This follow are currently not possible as the service has a `husky` script that breaks install. We need to look into creating a dist that we publish of just the types.

- Types need be imported from the service
- (If possible) import `MAX_HOUR_RELAYS` from the service (or load via eventual config)

Note: the max. number of relays is currently hardcoded in the service, but also accepts an env. var. We should expedite looking into setting this via a config service.

## Checklist
* [x] I've tested the branch on mobile 📱
* [x] I've documented how it affects the analytics (if at all) 📊
* [x] I've written a unit/e2e test for it (if applicable) 🧑‍💻

### CODE REVIEW

File: src/services/pairing/module.ts [hunk 1]
```diff
@@ -75,7 +75,6 @@ const pairingModule = (): WalletInit => {
             this.disconnected$ = new Subject()
             this.providers = {}
 
-            // @ts-expect-error - `payload` type (`ISessionStatus`) is not correctly `pipe`ed
```
Comment (iamacook) [inline src/services/pairing/module.ts:78]: The linter was complaining about these being unnecessary.

File: src/services/tx/relaying.ts [hunk 2]
```diff
@@ -0,0 +1,60 @@
+import { type SafeTransactionData } from '@safe-global/safe-core-sdk-types'
+import { IS_PRODUCTION, SAFE_RELAY_SERVICE_URL_PRODUCTION, SAFE_RELAY_SERVICE_URL_STAGING } from '@/config/constants'
+import { cgwDebugStorage } from '@/components/sidebar/DebugToggle'
+
+export const SAFE_RELAY_SERVICE_URL =
+  IS_PRODUCTION || cgwDebugStorage.get() ? SAFE_RELAY_SERVICE_URL_PRODUCTION : SAFE_RELAY_SERVICE_URL_STAGING
+
+// TODO: import types from relay-service
```
Comment (iamacook) [inline src/services/tx/relaying.ts:8]: Please see note in description.

**Comment by katspaugh** [comment]
Please rebase it onto #1842 as it already has some relayer refactors.

File: src/services/tx/relaying.ts (hunk 2 above)
Comment (iamacook) [reply src/services/tx/relaying.ts:8]: After discussion with @fmrsabino, this is going to be tackled as part of a Friday project - an auto-generated SDK.

File: src/hooks/useRelaysBySafe.ts [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+
+    let relays: RelayResponse[] | undefined
+
+    try {
+      relays = await Promise.all(ownerAddresses.map((address) => getRelays(chain.chainId, address)))
+    } catch (error) {
+      logError(Errors._630, (error as Error).message)
+    }
+
+    const mininmum = minBy(relays, 'remaining')
```
Comment (usame-algan) [inline src/hooks/useRelaysBySafe.ts:39]:
```suggestion
    const minimum = minBy(relays, 'remaining')
```

File: src/hooks/useRelaysBySafe.ts [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+
+    return mininmum || { remaining: 0, limit: MAX_HOUR_RELAYS }
```
Comment (usame-algan) [inline src/hooks/useRelaysBySafe.ts:41]:
```suggestion
    return minimum || { remaining: 0, limit: MAX_HOUR_RELAYS }
```

**Review by usame-algan — APPROVED** [review]

**Pull request merged**